src/workerd/api/queue.c++

```diff
@@ -523,6 +525,29 @@ kj::Promise<WorkerInterface::CustomEvent::Result> QueueCustomEventImpl::run(
   // reuse the scheduled worker logic and timeout for now.
   auto completed = co_await incomingRequest->finishScheduled();
 
+  // Log some debug info if the request did not complete fully:
+  // In particular, detect whether or not the users queue() handler function completed
+  // and include info about other waitUntil tasks that may have caused the request to timeout.
+  if (!completed) {
+    kj::String statusStr;
+    KJ_SWITCH_ONEOF(queueEventHolder->event->getCompletionStatus()) {
+      KJ_CASE_ONEOF(i, QueueEvent::Incomplete) {
+        statusStr = kj::str("Incomplete");
+        break;
+      }
+      KJ_CASE_ONEOF(s, QueueEvent::CompletedSuccessfully) {
+        statusStr = kj::str("Completed Succesfully");
+        break;
+      }
+      KJ_CASE_ONEOF(e, QueueEvent::CompletedWithError) {
+        statusStr = kj::str("Completed with error:", e.error);
+        break;
+      }
+    }
+    auto taskTrace = incomingRequest->getContext().getWaitUntilTasks().trace();
+    KJ_LOG(WARNING, "NOSENTRY queue event timed out. Status: ", statusStr, ". Tasks: ", taskTrace);
```

Comment:
FWIW I'd expect us to just want the following due to how KJ_LOG's formatting works:

```
KJ_LOG(WARNING, "NOSENTRY queue event timed out", statusStr, taskTrace);
```

But you can be the judge of that once you get your local test working.